pkg/internal/ebpf/generictracer/generictracer.go

```diff
@@ -287,64 +349,64 @@ func (p *Tracer) KProbes() map[string]ebpfcommon.FunctionPrograms {
 	}
 }
 
-func (p *Tracer) Tracepoints() map[string]ebpfcommon.FunctionPrograms {
+func (p *Tracer) Tracepoints() map[string]ebpfcommon.ProbeDesc {
 	return nil
 }
 
-func (p *Tracer) UProbes() map[string]map[string]ebpfcommon.FunctionPrograms {
-	return map[string]map[string]ebpfcommon.FunctionPrograms{
+func (p *Tracer) UProbes() map[string]map[string][]*ebpfcommon.ProbeDesc {
```

Comment:
OK, cool, so this was somewhere I thought we needed to go, having an array of probes on a single function when we need to deal with recompilation. Is this why you allow for this now, or there's another reason?

Reply:
Actually the reason was to normalise the code between `GoProbes()` and `UProbes()` - now they both return maps whose values are `map[string][]*ebpfcommon.ProbeDesc`, which enables the code to handle them to be generic in instrumenter.go (in particular, the `instrumentProbes()` function).